Runtimes/Supplemental/Synchronization/CMakeLists.txt

```
@@ -0,0 +1,121 @@
+
+option(${PROJECT_NAME}_INSTALL_NESTED_SUBDIR "Install libraries under a platform and architecture subdirectory"
+  ${SwiftCore_INSTALL_NESTED_SUBDIR})
+
+option(${PROJECT_NAME}_PLATFORM_SUBDIR "Platform name used for installed swift{doc,module,interface} files"
+  ${SwiftCore_PLATFORM_SUBDIR})
+
+option(${PROJECT_NAME}_COMPILER_VARIANT_TARGET "Compiler variant target to use when emiting modules"
+  ${SwiftCore_COMPILER_VARIANT_TARGET})
+
+include(CatalystSupport)
+include(EmitSwiftInterface)
+
+add_compile_options(
+  "$<$<COMPILE_LANGUAGE:Swift>:-nostdlibimport>"
+  "$<$<AND:$<BOOL:${${PROJECT_NAME}_ENABLE_LIBRARY_EVOLUTION}>,$<COMPILE_LANGUAGE:Swift>>:-enable-library-evolution>"
+  "$<$<AND:$<BOOL:${${PROJECT_NAME}_ENABLE_PRESPECIALIZATION}>,$<COMPILE_LANGUAGE:Swift>>:SHELL:-Xfrontend -prespecialize-generic-metadata>"
+  $<$<COMPILE_LANGUAGE:Swift>:-explicit-module-build>
+  $<$<COMPILE_LANGUAGE:Swift>:-enable-builtin-module>
+  $<$<COMPILE_LANGUAGE:Swift>:-strict-memory-safety>
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-enable-experimental-feature RawLayout>"
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-enable-experimental-feature StaticExclusiveOnly>"
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-enable-experimental-feature Extern>")
+
+gyb_expand(Atomics/AtomicIntegers.swift.gyb Atomics/AtomicIntegers.swift)
+gyb_expand(Atomics/AtomicStorage.swift.gyb Atomics/AtomicStorage.swift)
+
+add_library(swiftSynchronization
+  Atomics/Atomic.swift
+  Atomics/AtomicBool.swift
+  Atomics/AtomicFloats.swift
+  Atomics/AtomicLazyReference.swift
+  Atomics/AtomicMemoryOrderings.swift
+  Atomics/AtomicOptional.swift
+  Atomics/AtomicPointers.swift
+  Atomics/AtomicRepresentable.swift
+  Atomics/WordPair.swift
+  Atomics/AtomicStorage.swift
+  Atomics/AtomicIntegers.swift
+  Cell.swift
+  Mutex/Mutex.swift
+  $<$<PLATFORM_ID:Darwin>:Mutex/DarwinImpl.swift>
+  $<$<PLATFORM_ID:Linux>:Mutex/LinuxImpl.swift>
+  $<$<PLATFORM_ID:WASI>:Mutex/SpinLoopHint.swift>
+  $<$<PLATFORM_ID:WINDOWS>:Mutex/WindowsImpl.swift>)
+
+set_target_properties(swiftSynchronization PROPERTIES
+  Swift_MODULE_NAME Synchronization
+  Swift_COMPILATION_MODE wholemodule)
+
+if(APPLE AND BUILD_SHARED_LIBS)
+  target_link_options(swiftSynchronization PRIVATE "SHELL:-Xlinker -headerpad_max_install_names")
+endif()
+
+target_link_libraries(swiftSynchronization
+  PRIVATE
+    swiftCore
+    $<$<PLATFORM_ID:Darwin>:swiftDarwin>)
+
+set(${PROJECT_NAME}_INSTALL_LIBDIR "${CMAKE_INSTALL_LIBDIR}/swift$<$<NOT:$<BOOL:${BUILD_SHARED_LIBS}>>:_static>$<$<BOOL:${Supplemental_INSTALL_NESTED_SUBDIR}>:/${Supplemental_PLATFORM_SUBDIR}/${Supplemental_ARCH_SUBDIR}>" CACHE STRING "")
+set(${PROJECT_NAME}_INSTALL_SWIFTMODULEDIR "${CMAKE_INSTALL_LIBDIR}/swift$<$<NOT:$<BOOL:${BUILD_SHARED_LIBS}>>:_static>$<$<BOOL:${Supplemental_INSTALL_NESTED_SUBDIR}>:/${Supplemental_PLATFORM_SUBDIR}>" CACHE STRING "")
+
+install(TARGETS swiftSynchronization
+  EXPORT SwiftSupplementalTargets
+  COMPONENT SwiftCore_runtime
```

This seems a remnant from the reference code in SwiftCore -- maybe we can use the project name directly

```suggestion
  COMPONENT ${PROJECT_NAME}
```

---

Namespacing the component with the project name is good. Let's keep the `_runtime` suffix since this is only installing the runtime library. This will make it easier to split up the install story for different use cases.